### PR TITLE
rgw: user policy: forward write requests to master zone

### PR DESCRIPTION
**Pull request by theanalyst**
Since earlier releases didn't forward the requests, have an additional pass
through for delete policy requests to allow deletion when no policy is seen in
the master zone

### CODE REVIEW

**Comment by theanalyst** [comment]
only manually tested on a local cluster atm

File: src/rgw/rgw_rest_user_policy.cc [hunk 1]
```diff
@@ -131,6 +132,14 @@ void RGWPutUserPolicy::execute()
     return;
   }
 
+  if (!store->svc()->zone->is_meta_master()) {
+    ceph::bufferlist in_data;
+    op_ret = forward_request_to_master(s, nullptr, store, in_data, nullptr);
+    if (op_ret < 0) {
+      ldpp_dout(this, 0) << "forward_request_to_master returned ret=" << op_ret << dendl;
+    }
```
Comment (pritha-srivastava) [inline src/rgw/rgw_rest_user_policy.cc]: In case of error on master zone, do we still go ahead and create the policy in the other zone?

Comment (theanalyst) [reply src/rgw/rgw_rest_user_policy.cc]: sorry this code wasn't supposed to be here, I hadn't pushed the latest changes

Comment (theanalyst) [reply src/rgw/rgw_rest_user_policy.cc]: updated, for PUT requests, failure in master doesn't proceed for PUT in the local zone, the only exception is in delete requests where master returns a -ENOENT

**Comment by pritha-srivastava** [comment]
looks fine to me.

**Comment by theanalyst** [comment]
this passes s3-tests
run: http://pulpito.ceph.com/abhi-2020-01-24_16:05:12-rgw-wip-abhi-testing-2020-01-24-1135-distro-basic-smithi/ (which had s3-tests failure because of not being based on ceph-master)
rerun: http://pulpito.ceph.com/abhi-2020-01-27_10:30:43-rgw-wip-abhi-testing-2020-01-24-1135-distro-basic-smithi/

**Pull request merged**